common/interfaces.ts

```diff
@@ -16,6 +16,14 @@ export type Index = {
 /**
  ********** WORKFLOW TYPES/INTERFACES **********
  */
+export type MDSQueryParams = {
+  dataSourceId: string;
+};
+
+export interface MDSStates {
+  queryParams: MDSQueryParams;
+  selectedDataSourceId?: string;
+}
```

Comment:
I think this can be removed, or at the very least simplified. It is redundantly capturing `dataSourceId` both nested under `queryParams.dataSourceId` and in `selectedDataSourceId`.